### ISSUE TITLE
The C# compiler does not provide diagnostics about invalid explicit struct layouts.

### ISSUE THREAD

**Issue by vsfeedback** [dotnet/roslyn#23078]
Created a simple .Net Framework Console App. When I start debugging, I get a tab titled &quot;Break Mode&quot; with the header text &quot;The application is in break mode&quot;. There are no breakpoints in the application, and if I click &quot;Continue execution&quot; in this tab, none of my code executes and no actual breakpoints are hit.

_This issue has been moved from https://developercommunity.visualstudio.com/content/problem/142892/1574-the-application-is-in-break-mode-immediately.html
VSTS ticketId: 518718_
_These are the original issue comments:_

Hugh Phoenix-Hulme on 11/2/2017, 04:33 AM (6 days ago): The code that fails to run has a struct with LayoutKind.Explicit and overlapping fields.When the code contains the instantiation of the struct, it will not run - not even to the point that the struct is created. Removing the instantiation, the code will run. Removing the overlapping field, the code will run.My feeling is that this code should run - but if there is some reason it's invalid, the compiler should be reporting an error.(I don't want a technical explanation of why I'm totally bonkers to have a byte array overlapping an int and a string array - I want an error from the compiler telling me why this code won't run.)    

```csharp
using System;
using System.Runtime.InteropServices;

class Program
{
    static void Main()
    {
        Console.WriteLine("Hello World");
        NoseTypes t = new NoseTypes { Names = new[] { "Long", "Short", "Button" } };
    }
}

[StructLayout(LayoutKind.Explicit)]
public struct NoseTypes {
    [FieldOffset(0)] public int Size;
    [FieldOffset(4)] public string[] Names;
    [FieldOffset(0)] public byte[] Data;
}
```

Hugh Phoenix-Hulme on 11/2/2017, 04:35 AM (6 days ago): SORRY, 15.4.2 not 15.7.4. (Why is 2017 version 15? And how do I edit my initial report?)
_These are the original issue solutions:_
(no solutions)

**Comment by agocke** [comment]
This generally appears to be by-design. The C# compiler has never provided diagnostics around invalid explicit layouts, including the native compiler.

To do so now would be a breaking change, but would it be reasonable to do so if the resulting program fails to type load anyway? The CLR immediately produces an error on type load saying:

```
Could not load type 'NoseTypes' from assembly 'ConsoleApp1, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null' because it contains an object field at offset 0 that is incorrectly aligned or overlapped by a non-object field.
```

**Comment by agocke** [comment]
cc @jaredpar

**Comment by agocke** [comment]
On the other hand, it seems that if the type is not used, the program will run without error in the CLR, so maybe this change would be worse than I thought.

**Comment by gafter** [comment]
For the reasons indicated above, this is not something we would want to add warnings or errors for.
